app/controllers/api_controller.rb

```diff
@@ -40,9 +40,15 @@ def doorkeeper_application_owner
   end
 
   def user_for_paper_trail
+    return headers['X-Current-User'] if headers['X-Current-User'].present?
+
     current_user.owner_id
   end
 
+  def info_for_paper_trail
+    { user_type: headers['X-Current-User'].present? ? 'user' : 'supplier' }
```

Comment:
unfortunately, this is not a safe assumption. Instead use the `doorkeeper_application_owner` in the ApiController. This will be nil for the front end or the supplier (geo / serco) for an api client.